src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1204,7 +1204,7 @@ class TcgStatics {
     if(
       (potentialFossil.cardTypes.is(ITEM) && potentialFossil.name.contains("Fossil")) ||
       (potentialFossil.cardTypes.is(STAGE1) && potentialFossil.predecessor.contains("Fossil")) ||
-      (potentialFossil.cardTypes.is(STAGE2) && bg.gm().getBasicsFromStage2(potentialFossil.name).any{it.contains("Fossil")})
+      (potentialFossil.cardTypes.is(STAGE2) && bg().gm().getBasicsFromStage2(potentialFossil.name).any{it.contains("Fossil")})
```

Comment:
My understanding at this point is that `bg` doesn't require parens because it is a top-level Groovy method in the `TcgStatics` however `gm()` (and `em()`) does as it is a method on a Java `Battleground` class that `bg` returns. So this change shouldn't be necessary, but doesn't hurt either.

Reply:
Yeah, doing it just to try (and to avoid doing another commit/PR in case it _was_ needed, heh). As you said, doesn't hurt :P

One "clue" I found about this is that the filter works fine on discard (where I use filterAll). The crash happens exclusively on the search menu, and only on STAGE2 cards. No matter if fossil or not, that's the area where it breaks and gives [this crash](https://gist.githubusercontent.com/starg09/0611c4b5e80e68bc3a0f5b1509cef6b1/raw/7aefb03d7d332e63d24cd29c709b7a3bcc067aad/Excavator_Search_Stage2_Crash)...

Reply:
Interestingly, I think with how Groovy works this line could actually be
```
(potentialFossil.cardTypes.is STAGE2 && bg.gm().getBasicsFromStage2 potentialFossil.name findAll {it.contains "Fossil"})
``` 

from some of the Groovy stuff I've been looking up. Not that I think you should stray _too far_ from what seems to be standard currently.